pyface/ui/qt4/tasks/advanced_editor_area_pane.py

```diff
@@ -644,6 +684,10 @@ def __init__(self, editor_widget):
 
         self.tabCloseRequested.connect(editor_widget.editor.close)
 
+    def disconnect_event_listeners(self, editor_widget):
+        # Assumes that the editor of editor_widget never changes
+        self.tabCloseRequested.disconnect(editor_widget.editor.close)
```

Comment:
I agree this one is riskier: Disconnecting something that isn't connected will cause Qt to raise an error. I'd be in favour of leaving this one out.